### DISCUSSION TITLE
Map types from composer vendor namespace

### DISCUSSION THREAD

**Discussion by ilmDitsch** [thecodingmachine/graphqlite#625] [Q&A] [answered]
Hello,

I have been using the library for some time now and would like to take this opportunity to thank you for the great work.

Since a short time we use a private composer repository to manage our models. They are provided with PHP attributes there. When I include this library in our API project as a dependency, some types can no longer be mapped.

We're using a pretty normal setup with `SchemaFactory`:

```php
$factory = new SchemaFactory($cache, $container);
$factory->addTypeNamespace('Namespace\\From\\ComposerLib\\In\\VendorDir');
return $factory->createSchema();
```

As far as I understand it, it could be due to [`GlobTypeMapper`](https://github.com/thecodingmachine/graphqlite/blob/master/src/Mappers/GlobTypeMapper.php) not taking into account namespaces in the vendor directory. Is it possible to configure GraphQLite for this scenario? 

Thanks a lot in advance!

**Comment by oojacoboo** [comment]
Hey @ilmDitsch.  I'm not aware of any limitations here.  You need to be sure that lib is autoloaded properly with composer.  Also, I don't believe the double `\\` is necessary here.  Lastly, I'm going to assume you're using the proper namespace that can successfully be referenced in other autoloaded files.

**Reply by ilmDitsch** [reply]
The lib is autoloaded and available within the project. We're using other classes from it within the code. I tried all variants in the `addTypeNamespace` method, with and without escaped `\`...

As mentioned, the doc block of [`GlobTypeMapper`](https://github.com/thecodingmachine/graphqlite/blob/master/src/Mappers/GlobTypeMapper.php) states, that the vendor folder is not scanned:
https://github.com/thecodingmachine/graphqlite/blob/3041608bc6e46fd633f2c823f20cedc0b24adee7/src/Mappers/GlobTypeMapper.php#L20

As far as I understood the implementation of the `SchemaFactory`, this type mapper is used to extract annotations (also PHP attributes?) from a namespace.

**Reply by oojacoboo** [reply, marked as answer]
@ilmDitsch well, I guess that's pretty clear :)  I can only assume this is being done for performance reasons.  Being that namespaces are explicitly defined, I don't see any reason not to support the vendor directory as well, if you'd like to submit a PR on this.
